### PR TITLE
Fix/521 pdf2image memory error hi res

### PR DESCRIPTION
**Pull request by christinestraub**
### Summary
This PR is to reflect changes in the `unstructured-inference` [PR #152](https://github.com/Unstructured-IO/unstructured-inference/pull/152)

- Update functionality to retrieve image metadata from a page for `document_to_element_list`

### Testing

- Run the following command:

```
$ cd unstructured
$ pip uninstall unstructured-inference
$ git clone -b fix/pdf2image-too-many-open-files https://github.com/Unstructured-IO/unstructured-inference.git
$ cd unstructured-inference
$ pip install -e .
```

- Run the following code and measure the memory usage.

```
filename = "example-docs/pdf2image-memory-error-test-400p.pdf"
elements = partition_pdf(filename, strategy="hi_res")
print("\n\n".join([str(el) for el in elements]))
````

### CODE REVIEW

**Review by MthwRobinson — APPROVED** [review]
This looks good to me, but probably worth a look from @qued as well since it involves the coordinate system.

File: unstructured/file_utils/filetype.py [hunk 1]
```diff
@@ -466,14 +466,20 @@ def document_to_element_list(
     num_pages = len(document.pages)
     for i, page in enumerate(document.pages):
         page_elements: List[Element] = []
+
+        page_image_metadata = _get_page_image_metadata(page)
+        image_format = page_image_metadata.get("format")
```
Comment (benjats07) [inline unstructured/file_utils/filetype.py:471]: Wondering if this ever returns None (because the keys always exist)

**Review by benjats07 — APPROVED** [review]

**Review by benjats07 — CHANGES_REQUESTED** [review]
Well, fixed the memory error, but now throws:

AttributeError                            Traceback (most recent call last)
Cell In[6], line 1
----> 1 elements = partition_pdf(filename, strategy="hi_res")

File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/documents/elements.py:214, in process_metadata.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
    212 @wraps(func)
    213 def wrapper(*args, **kwargs):
--> 214     elements = func(*args, **kwargs)
    215     sig = inspect.signature(func)
    216     params = dict(**dict(zip(sig.parameters, args)), **kwargs)

File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/file_utils/filetype.py:531, in add_metadata_with_filetype.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
    529 @wraps(func)
    530 def wrapper(*args, **kwargs):
--> 531     elements = func(*args, **kwargs)
    532     sig = inspect.signature(func)
    533     params = dict(**dict(zip(sig.parameters, args)), **kwargs)

File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/partition/pdf.py:85, in partition_pdf(filename, file, include_page_breaks, strategy, infer_table_structure, ocr_languages, max_partition, include_metadata, metadata_filename, **kwargs)
     55 """Parses a pdf document into a list of interpreted elements.
     56 Parameters
     57 ----------
   (...)
     82     no maximum is applied. Only applies to the "ocr_only" strategy.
     83 """
     84 exactly_one(filename=filename, file=file)
---> 85 return partition_pdf_or_image(
     86     filename=filename,
     87     file=file,
     88     include_page_breaks=include_page_breaks,
     89     strategy=strategy,
     90     infer_table_structure=infer_table_structure,
     91     ocr_languages=ocr_languages,
     92     max_partition=max_partition,
     93     **kwargs,
     94 )

File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/partition/pdf.py:151, in partition_pdf_or_image(filename, file, is_image, include_page_breaks, strategy, infer_table_structure, ocr_languages, max_partition, **kwargs)
    149     with warnings.catch_warnings():
    150         warnings.simplefilter("ignore")
--> 151         layout_elements = _partition_pdf_or_image_local(
    152             filename=filename,
    153             file=spooled_to_bytes_io_if_needed(file),
    154             is_image=is_image,
    155             infer_table_structure=infer_table_structure,
    156             include_page_breaks=include_page_breaks,
    157             ocr_languages=ocr_languages,
    158             **kwargs,
    159         )
    161 elif strategy == "fast":
    162     return extracted_elements

File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/utils.py:43, in requires_dependencies.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
     34 if len(missing_deps) > 0:
     35     raise ImportError(
     36         f"Following dependencies are missing: {', '.join(missing_deps)}. "
     37         + (
   (...)
     41         ),
     42     )
---> 43 return func(*args, **kwargs)

File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/partition/pdf.py:228, in _partition_pdf_or_image_local(filename, file, is_image, infer_table_structure, include_page_breaks, ocr_languages, model_name, **kwargs)
    220 else:
    221     layout = process_data_with_model(
    222         file,
    223         is_image=is_image,
   (...)
    226         model_name=model_name,
    227     )
--> 228 elements = document_to_element_list(layout, include_page_breaks=include_page_breaks, sort=False)
    229 out_elements = []
    231 for el in elements:

File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/file_utils/filetype.py:455, in document_to_element_list(document, include_page_breaks, sort)
    453 for layout_element in page.elements:
    454     if hasattr(page, "image") and hasattr(layout_element, "coordinates"):
--> 455         image_format = page.image.format
    456         coordinate_system = PixelSpace(width=page.image.width, height=page.image.height)
    457     else:

AttributeError: 'NoneType' object has no attribute 'format'

**Comment by christinestraub** [comment]
> Well, fixed the memory error, but now throws:
> 
> AttributeError Traceback (most recent call last) Cell In[6], line 1 ----> 1 elements = partition_pdf(filename, strategy="hi_res")
> 
> File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/documents/elements.py:214, in process_metadata..decorator..wrapper(*args, **kwargs) 212 @wraps(func) 213 def wrapper(*args, **kwargs): --> 214 elements = func(*args, **kwargs) 215 sig = inspect.signature(func) 216 params = dict(**dict(zip(sig.parameters, args)), **kwargs)
> 
> File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/file_utils/filetype.py:531, in add_metadata_with_filetype..decorator..wrapper(*args, **kwargs) 529 @wraps(func) 530 def wrapper(*args, **kwargs): --> 531 elements = func(*args, **kwargs) 532 sig = inspect.signature(func) 533 params = dict(**dict(zip(sig.parameters, args)), **kwargs)
> 
> File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/partition/pdf.py:85, in partition_pdf(filename, file, include_page_breaks, strategy, infer_table_structure, ocr_languages, max_partition, include_metadata, metadata_filename, **kwargs) 55 """Parses a pdf document into a list of interpreted elements. 56 Parameters 57 ---------- (...) 82 no maximum is applied. Only applies to the "ocr_only" strategy. 83 """ 84 exactly_one(filename=filename, file=file) ---> 85 return partition_pdf_or_image( 86 filename=filename, 87 file=file, 88 include_page_breaks=include_page_breaks, 89 strategy=strategy, 90 infer_table_structure=infer_table_structure, 91 ocr_languages=ocr_languages, 92 max_partition=max_partition, 93 **kwargs, 94 )
> 
> File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/partition/pdf.py:151, in partition_pdf_or_image(filename, file, is_image, include_page_breaks, strategy, infer_table_structure, ocr_languages, max_partition, **kwargs) 149 with warnings.catch_warnings(): 150 warnings.simplefilter("ignore") --> 151 layout_elements = _partition_pdf_or_image_local( 152 filename=filename, 153 file=spooled_to_bytes_io_if_needed(file), 154 is_image=is_image, 155 infer_table_structure=infer_table_structure, 156 include_page_breaks=include_page_breaks, 157 ocr_languages=ocr_languages, 158 **kwargs, 159 ) 161 elif strategy == "fast": 162 return extracted_elements
> 
> File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/utils.py:43, in requires_dependencies..decorator..wrapper(*args, **kwargs) 34 if len(missing_deps) > 0: 35 raise ImportError( 36 f"Following dependencies are missing: {', '.join(missing_deps)}. " 37 + ( (...) 41 ), 42 ) ---> 43 return func(*args, **kwargs)
> 
> File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/partition/pdf.py:228, in _partition_pdf_or_image_local(filename, file, is_image, infer_table_structure, include_page_breaks, ocr_languages, model_name, **kwargs) 220 else: 221 layout = process_data_with_model( 222 file, 223 is_image=is_image, (...) 226 model_name=model_name, 227 ) --> 228 elements = document_to_element_list(layout, include_page_breaks=include_page_breaks, sort=False) 229 out_elements = [] 231 for el in elements:
> 
> File ~/.pyenv/versions/3.8.17/envs/unstructured/lib/python3.8/site-packages/unstructured/file_utils/filetype.py:455, in document_to_element_list(document, include_page_breaks, sort) 453 for layout_element in page.elements: 454 if hasattr(page, "image") and hasattr(layout_element, "coordinates"): --> 455 image_format = page.image.format 456 coordinate_system = PixelSpace(width=page.image.width, height=page.image.height) 457 else:
> 
> AttributeError: 'NoneType' object has no attribute 'format'

@benjats07 Where did you get this error? It seems like the CI error for the `unstructured-inference` PR #152.

**Review by benjats07 — APPROVED** [review]

File: unstructured/file_utils/filetype.py (hunk 1 above)
Comment (christinestraub) [reply unstructured/file_utils/filetype.py:471]: The `image_format` would be `None` when you run this test - https://github.com/Unstructured-IO/unstructured/blob/fix/521-pdf2image-memory-error-hi-res/test_unstructured/file_utils/test_filetype.py#L475

**Pull request merged**